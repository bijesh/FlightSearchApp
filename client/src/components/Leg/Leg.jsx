import React from 'react';
import BpkLargeRightArrowIcon from 'bpk-component-icon/lg/long-arrow-right';
import BpkText from 'bpk-component-text';
import STYLES from './Leg.scss';

const legStyle = className => STYLES[className] || 'UNKNOWN';

const getStops=(stops)=>{
    return stops>0 ? `${stops} Stop`:'Direct';
};

const getTimeFromDateTime = (dateTime) => {
    return dateTime.substr(dateTime.indexOf('T') + 1, 5);
  };
const findDurationInHours = duration => `${Math.round(duration / 60)}h ${duration % 60}`;

const Leg = React.memo(({ leg }) => (
   <div>
   <div className={legStyle('Leg__img-parent')}><img alt="" src={leg.ImageUrl} /></div>
   <div className={legStyle('Leg__origin')}>
      <BpkText tagName="p">{getTimeFromDateTime(leg.DepartureTime)}</BpkText>
      <BpkText className={legStyle('Leg__origin__place')} tagName="p">{leg.DepAirport}</BpkText>
    </div> 
    <div className={legStyle('Leg__arrow')}>
      <BpkLargeRightArrowIcon className={legStyle('Leg__rightArrowIcon')} />
    </div>
    <div className={legStyle('Leg__destination')}>
      <BpkText tagName="p">{getTimeFromDateTime(leg.ArrivalTime)}</BpkText>
      <BpkText className={legStyle('Leg__origin__place')} tagName="p">{leg.ArvAirport}</BpkText>
    </div>

    <div className={legStyle('Leg__length')}>
       <div className={legStyle('Leg__duration')}>{findDurationInHours(leg.Duration)}</div>
        <div className={legStyle('Leg__stops')}>{getStops(leg.Stops)}</div>
     </div>

    </div>

));

export default Leg;