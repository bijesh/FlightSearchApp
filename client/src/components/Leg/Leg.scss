@import '~bpk-mixins';

.Leg {
    &__main {
        margin: 3%;
    }

    @media all and (max-width: 480px) {
        &__img-parent {
            height: 32px;
            width: 10%;
            border-radius: 3px;
            float: left;
            margin: 6% 34px 0 0;
        }
        &__img-parent img {
            height: 100%;
            width: 100%;
            float: left;
        }
        &__origin {
            width: 20%;
            float: left;
            &__place {
                color: #A6A6A6;
            }
        }

        &__destination {
            width: 18%;
            align-content: flex-start;
            float: left;
        }

        &__arrow {
            width: 20%;
            align-content: center;
            float: left;
        }

        &__stops {
            width: 60%;
            text-align: right;
            color: green;
            float: left;
        }
        &__duration {
            width: 60%;
            text-align: right;
            color: #A6A6A6;
            float: left;
        }
        &__length {
            float: right
        }
        &__rightArrowIcon {
            fill: #A6A6A6;
        }
    }
}