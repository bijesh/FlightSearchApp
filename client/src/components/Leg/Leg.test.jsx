import React from 'react';
import TestRenderer from 'react-test-renderer';
import Leg from './Leg';

const LegData = {
  ImageUrl:'EZY.png',
  DepartureTime: '2019-06-14T22:25:00',
  DepAirport:'LHR',
  ArrivalTime : '2019-06-14T22:25:00',
  ArvAirport : 'EDI',
  Duration:100,
  Stops : 1
};

describe('Leg', () => {
  it('should render correctly', () => {
    const tree = TestRenderer.create(<Leg leg={LegData}/>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

