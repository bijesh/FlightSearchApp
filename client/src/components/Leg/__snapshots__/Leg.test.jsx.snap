// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Leg should render correctly 1`] = `
<div>
  <div
    className="Leg__img-parent"
  >
    <img
      alt=""
      src="EZY.png"
    />
  </div>
  <div
    className="Leg__origin"
  >
    <p
      className="bpk-text bpk-text--base"
    >
      22
    </p>
    <p
      className="bpk-text bpk-text--base Leg__origin__place"
    >
      LHR
    </p>
  </div>
  <div
    className="Leg__arrow"
  >
    <svg
      className="Leg__rightArrowIcon"
      height="24"
      style={
        Object {
          "height": "1.5rem",
          "width": "1.5rem",
        }
      }
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M14.4 19.5l5.7-5.3c.4-.4.7-.9.8-1.5.1-.3.1-.5.1-.7s0-.4-.1-.6c-.1-.6-.4-1.1-.8-1.5l-5.7-5.3c-.8-.8-2.1-.7-2.8.1-.8.8-.7 2.1.1 2.8l2.7 2.5H5c-1.1 0-2 .9-2 2s.9 2 2 2h9.4l-2.7 2.5c-.5.4-.7 1-.7 1.5s.2 1 .5 1.4c.8.8 2.1.8 2.9.1z"
      />
    </svg>
  </div>
  <div
    className="Leg__destination"
  >
    <p
      className="bpk-text bpk-text--base"
    >
      22
    </p>
    <p
      className="bpk-text bpk-text--base Leg__origin__place"
    >
      EDI
    </p>
  </div>
  <div
    className="Leg__length"
  >
    <div
      className="Leg__duration"
    >
      100
    </div>
    <div
      className="Leg__stops"
    >
      1 Stop
    </div>
  </div>
</div>
`;
